### PR TITLE
Spec. analysis cleanup + tracker analysis

### PR DESCRIPTION
**Pull request by antsr**
Cleans up (removes) old spectroscopy analysis_v2 code.

Implements new spectroscopy base analysis class that can handle:
1d sweeps, 2d sweeps in 1 file, a combination of the previous in multiple files.
The base analysis handles combining the multiple datasets and can do pca.

Currently the only specific analysis that uses the new base class is the new QubitTrackerSpectroscopy class.
This class fits qubit spec pca data to Gaussians and can provides sweep ranges for the next sweep rectangle by fitting the peak positions to a polynomial model.

Minor change is to make base analysis class clean the fig. and ax. objects when closing objects such that the figures can be recreated in the notebook.

FYI @stephlazar

### CODE REVIEW

File: pycqed/analysis_v2/spectroscopy_analysis.py [hunk 1]
```diff
@@ -31,1507 +23,285 @@ def __init__(self, t_start: str = None,
...
-        else:
-            self.fit_dicts['reso_fit'] = {'fit_fn': fit_fn,
-                                          'fit_guess_fn': fit_guess_fn,
-                                          'fit_yvals': [{'data': np.squeeze(tt)} for tt in self.plot_amp],
-                                          'fit_xvals': np.squeeze([{'f': tt[0]} for tt in self.plot_frequency])}
-    def analyze_fit_results(self):
-        pass
-
-
-class ResonatorSpectroscopy(Spectroscopy):
```
Comment (stephlazar) [inline pycqed/analysis_v2/spectroscopy_analysis.py:413]: You cannot remove this, it is being used in QuDev_transmon.find_readout_frequency(). This is the routine we use to find the optimal readout frequency.

File: pycqed/analysis_v2/spectroscopy_analysis.py [hunk 2]
```diff
@@ -31,1507 +23,285 @@ def __init__(self, t_start: str = None,
...
-
-    def plot(self, key_list=None, axs_dict=None, presentation_mode=None, no_label=False):
-        super(ResonatorSpectroscopy, self).plot(key_list=key_list,
-                                                axs_dict=axs_dict,
-                                                presentation_mode=presentation_mode)
-        if self.do_fitting:
-            self.plot_fitting()
-
-
-class ResonatorSpectroscopy_v2(Spectroscopy):
```
Comment (stephlazar) [inline pycqed/analysis_v2/spectroscopy_analysis.py:800]: You cannot remove this, it is being used in QuDev_transmon.find_readout_frequency(). This is the routine we use to find the optimal readout frequency.

**Review by stephlazar — CHANGES_REQUESTED** [review]
Nice to have this addition, thanks for taking the time to write it.
But please add back the two classes you deleted.

File: pycqed/analysis_v2/spectroscopy_analysis.py (hunk 2 above)
Comment (stephlazar) [reply pycqed/analysis_v2/spectroscopy_analysis.py:800]: Also check that these classes still work after your changes to BaseDataAnalysis and other changes in spectroscopy_analysis.py

Comment (stephlazar) [reply pycqed/analysis_v2/spectroscopy_analysis.py:800]: Finally, did you check that the changes in BaseDataAnalysis do not break MultiQubit_TimeDomain_Analysis?

**Comment by stephlazar** [comment]
Also check that these classes still work after your changes to BaseDataAnalysis and other changes in spectroscopy_analysis.py

**Comment by stephlazar** [comment]
Finally, did you check that the changes in BaseDataAnalysis do not break MultiQubit_TimeDomain_Analysis?

**Review by chellings — CHANGES_REQUESTED** [review]
Thanks to @stephlazar for volunteering to review since she can judge this better than I can. In addition to her comments, I have spotted some missing docstrings in the new code in spectroscopy_analysis.py, so it would be nice to also add those.

File: pycqed/analysis_v2/base_analysis.py [hunk 3]
```diff
@@ -634,8 +635,16 @@ def save_figures(self, savedir: str = None, savebase: str = None,
...
+        if close_figs:
+            self.close_figs(key_list)
+
+    def close_figs(self, key_list='auto'):
+        if key_list == 'auto' or key_list is None:
+            key_list = self.figs.keys()
+        for key in list(key_list):
+            plt.close(self.figs[key])
+            self.figs.pop(key)
+            self.axs.pop(self.plot_dicts[key]['fig_id'])
```
Comment (chellings) [inline pycqed/analysis_v2/base_analysis.py]: @antsr This change makes CryoscopeAnalysis crash

Comment (antsr) [reply pycqed/analysis_v2/base_analysis.py]: I fixed and tested it.

**Comment by antsr** [comment]
Hey, I hopefully addressed now all of the issues you raised @stephlazar @chellings. You could have another look now to see if we can get this merged. If you have any further tests you want me to do, let me know.

**Comment by antsr** [comment]
> Finally, did you check that the changes in BaseDataAnalysis do not break MultiQubit_TimeDomain_Analysis?

I tested that I was able to run the `CryoscopeAnalysis`, which is based on the `MultiQubit_TimeDomain_Analysis` with the new changes. Do you think I should try some other analysis as well?

Furthermore, I tested that I can call `MA.plot()` in the jupyter notebook and see the generated plots. After calling `MA.close_figs()`, I can change the plot_dicts and rerun `MA.plot()` to see updated figures. This was the use case that I needed the changes for.

**Review by chellings — COMMENTED** [review]
My comments have been addressed. I would be fine with merging if @stephlazar approves the changes. As in the first round of reviews, I have not looked at everything since steph is more familiar with the analysis code.

I have merged the branch into Proj/S17 (pushed to github) so that it can be used on a (real or virtual) setup for a while to see whether the analysis changes break anything. Maybe @stephlazar can run a few standard analysis calls as part of doing her review. Thank you!

**Review by chellings — APPROVED** [review]

**Review by stephlazar — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**